File: python/python_b2b_seeding/python_service/database/models/tenant.py
"""
Tenant model.
Similar to the Node.js tenant.model.ts implementation.
"""
from sqlalchemy import Column, String, Boolean, Text, JSON, DateTime
from sqlalchemy.orm import relationship
from database.models.base import BaseModel
from database.models.types import GUID
import uuid


class Tenant(BaseModel):
    """Tenant model."""
    __tablename__ = 'tenants'
    
    # Explicitly define id first to ensure it appears first in database
    id = Column(GUID(), primary_key=True, default=uuid.uuid4)
    
    name = Column(String(100), nullable=False)
    code = Column(String(50), unique=True, nullable=False, index=True)
    domain = Column(String(255), nullable=True)
    description = Column(Text, nullable=True)
    industry = Column(String(100), nullable=True)
    website_url = Column(String(255), name='websiteUrl', nullable=True)
    logo_url = Column(String(255), name='logoUrl', nullable=True)
    contact_email = Column(String(100), name='contactEmail', nullable=True)
    contact_phone = Column(String(20), name='contactPhone', nullable=True)
    address = Column(String(100), nullable=True)
    city = Column(String(50), nullable=True)
    state = Column(String(50), nullable=True)
    postal_code = Column(String(20), name='postalCode', nullable=True)
    country = Column(String(50), nullable=True)
    
    status = Column(String(20), default='active', nullable=False)
    is_active = Column(Boolean, name='isActive', default=True, nullable=False)
    is_verified = Column(Boolean, name='isVerified', default=False, nullable=False)
    
    settings = Column(JSON, nullable=True)
    billing_config = Column(JSON, name='billingConfig', nullable=True)
    
    verified_at = Column(DateTime, name='verifiedAt', nullable=True)
    verified_by = Column(String(255), name='verifiedBy', nullable=True)
    
    deleted_at = Column(DateTime, name='deletedAt', nullable=True)
    
    # Relationships
    users = relationship("User", back_populates="tenant")

File: dotnet/dotnet_b2b_seeding/dotnet_service/Database/Models/Tenant.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json;

namespace DotnetService.Database.Models
{
    /// <summary>
    /// Tenant model.
    /// Similar to the Python tenant.py implementation.
    /// </summary>
    [Table("tenants")]
    public class Tenant : BaseModel
    {
        [Required]
        [MaxLength(100)]
        [Column("name")]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(50)]
        [Column("code")]
        public string Code { get; set; } = string.Empty;
        
        [MaxLength(255)]
        [Column("domain")]
        public string? Domain { get; set; }
        
        [Column("description", TypeName = "text")]
        public string? Description { get; set; }
        
        [MaxLength(100)]
        [Column("industry")]
        public string? Industry { get; set; }
        
        [MaxLength(255)]
        [Column("websiteUrl")]
        public string? WebsiteUrl { get; set; }
        
        [MaxLength(255)]
        [Column("logoUrl")]
        public string? LogoUrl { get; set; }
        
        [MaxLength(100)]
        [Column("contactEmail")]
        public string? ContactEmail { get; set; }
        
        [MaxLength(20)]
        [Column("contactPhone")]
        public string? ContactPhone { get; set; }
        
        [MaxLength(100)]
        [Column("address")]
        public string? Address { get; set; }
        
        [MaxLength(50)]
        [Column("city")]
        public string? City { get; set; }
        
        [MaxLength(50)]
        [Column("state")]
        public string? State { get; set; }
        
        [MaxLength(20)]
        [Column("postalCode")]
        public string? PostalCode { get; set; }
        
        [MaxLength(50)]
        [Column("country")]
        public string? Country { get; set; }
        
        [Required]
        [MaxLength(20)]
        [Column("status")]
        public string Status { get; set; } = "active";
        
        [Column("isActive")]
        public bool IsActive { get; set; } = true;
        
        [Column("isVerified")]
        public bool IsVerified { get; set; } = false;
        
        [Column("settings", TypeName = "json")]
        public string? Settings { get; set; }
        
        [Column("billingConfig", TypeName = "json")]
        public string? BillingConfig { get; set; }
        
        [Column("verifiedAt")]
        public DateTime? VerifiedAt { get; set; }
        
        [MaxLength(255)]
        [Column("verifiedBy")]
        public string? VerifiedBy { get; set; }
        
        [Column("deletedAt")]
        public DateTime? DeletedAt { get; set; }
        
        // Relationships
        public virtual ICollection<User> Users { get; set; } = new List<User>();
    }
}

